drivers/sdhc/rcar_mmc.c

```diff
@@ -1707,12 +1707,13 @@ static int rcar_mmc_init(const struct device *dev)
 	ret = pinctrl_apply_state(cfg->pcfg, PINCTRL_STATE_DEFAULT);
 	if (ret < 0) {
 		LOG_ERR("%s: error can't apply pinctrl state", dev->name);
-		return ret;
+		goto _unmap;
 	}
 
 	if (!device_is_ready(cfg->cpg_dev)) {
 		LOG_ERR("%s: error cpg_dev isn't ready", dev->name);
-		return -ENODEV;
+		ret = -ENODEV;
```

Comment:
(not your issue, minor): Looks like here we have to return -EBUSY instead of -ENODEV

Reply:
unfortunately, in this file all issues mine :)

Reply:
Yes, you are right, in most cases, in this case, functions return -ENODEV or less frequently  -EINVAL. So you can ignore my comment:) Thank you for your answer.